### PR TITLE
Added report writing from array of entries

### PR DESCRIPTION
**Pull request by sinarest1608**
This adds the functionality of logging to report using an array of entries. Also added the Unit tests. 

Refer #7 

To Review:
Check the code flow and verify that the unit tests are passed.

Note: It would be better to merge #9 and #10 before merging this.

### CODE REVIEW

**Comment by sinarest1608** [comment]
@mbentz-uf 

The lint workflow outputs the following E226 errors:
```python
./tests/report_handler/test_report_handler.py:59:47: E226 missing whitespace around arithmetic operator
./tests/report_handler/test_report_handler.py:66:35: E226 missing whitespace around arithmetic operator
./tests/report_handler/test_report_handler.py:76:47: E226 missing whitespace around arithmetic operator
./tests/report_handler/test_report_handler.py:91:35: E226 missing whitespace around arithmetic operator
```
But the file `./tests/report_handler/test_report_handler.py` does not exists on this branch. 
I fixed E226 on other files present on this branch in the previous commit.

File: src/report_handler/report_handler.py [hunk 1]
```diff
@@ -40,6 +40,31 @@ def add_entry_to_sheet(self, sheet, entry):
         else:
             self.logs[sheet].append(entry)
 
+    def add_data_to_sheet(self, sheet: str, data: dict, path: str):
+        if (not utils.containsKey(data, "headers") and not utils.containsKey(data, "rows")):
+            return Exception("Data signature mis-match! Data should have 'headers' and 'rows'")
```
Comment (mbentz-uf) [inline src/report_handler/report_handler.py:45]: Small typo, should be mismatch instead.

File: src/report_handler/report_handler.py [hunk 2]
```diff
@@ -40,6 +40,31 @@ def add_entry_to_sheet(self, sheet, entry):
...
+
+        headers, rows = data["headers"], data["rows"]
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+        file = self.prevent_overwrite(
+            f"{path}/rejected-publications-report-{datetime.today().strftime('%Y-%m-%d')}.xlsx")
```
Comment (mbentz-uf) [inline src/report_handler/report_handler.py:52]: I didn't do a good job explaining this in the issue. The function `add_to_sheet` is meant to add entries to the internal `log` variable (use `add_entry_to_sheet`), not to generate a separate xlsx. The code snippet I included in the example, is an example of how we will incorporate the new feature into ufvivo_pubingest.

**Review by mbentz-uf — CHANGES_REQUESTED** [review]
Please rewrite the function to add entries to the internal `log` variable and remove path from the function signature. To generate the xlsx `write_report` would still need to be called.

**Comment by mbentz-uf** [comment]
Do the tests show up for you in VSCode?

**Comment by sinarest1608** [comment]
@mbentz-uf 
I can't see the tests in VsCode. I remember using command line to run the tests.

**Comment by mbentz-uf** [comment]
Can you help me look into that? Removing the -s flag from `settings.json` fixes it but we use that flag for all of our other python projects. This is what the Output>Python:

image

**Comment by sinarest1608** [comment]
Changing the next argument after `-s` to `.` fixed this. 

```python
    "python.testing.unittestArgs": [
        "-v",
        "-s",
        ".",
        "-p",
        "*test*.py"
    ],
```

**Review by mbentz-uf — APPROVED** [review]

**Comment by mbentz-uf** [comment]
Looks good, thanks for the help with this!

**Pull request merged**